splunk_connect_for_snmp_mib_server/profiles.py

```diff
@@ -0,0 +1,31 @@
+#   ########################################################################
+#   Copyright 2021 Splunk Inc.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+#   ########################################################################
+
+import os
+
+import yaml
+
+
+def merge_profiles(directory, root_name):
+    result = {}
+    merged_profiles = {}
+    for root, directories, files in os.walk(directory, topdown=False):
```

Comment:
dict update() overwrites existing key, os.walk lists files in arbitrary order 
together these 2 facts can lead to unexpected overwriting of some profiles if they names will be the same

So running the code twice on same data can produce different results.

to fix that we will have to change code to
`for name in sorted(files)`